pkg/R/sparkR.R

```diff
@@ -149,11 +156,14 @@ sparkR.init <- function(
     }
     f <- file(path, open='rb')
     sparkRBackendPort <- readInt(f)
+    monitorPort <- readInt(f)
     close(f)
     file.remove(path)
-    if (length(sparkRBackendPort) == 0) {
+    if (length(sparkRBackendPort) == 0 || sparkRBackendPort == 0 ||
+        length(monitorPort) == 0 || monitorPort == 0) {
       stop("JVM failed to launch")
     }
+    assign(".monitorConn", socketConnection(port = monitorPort), envir = .sparkREnv)
```

Comment:
I think we should still have reg.finalizer to call sparkR.stop(), it's better to shutdown SparkContext gracefully (call SparkContext.stop()).